### PR TITLE
Implemented notification display component

### PR DESCRIPTION
**Pull request by Ilithor**

### CODE REVIEW

File: server/handlers/comment.js [hunk 1]
```diff
@@ -1,10 +1,33 @@
...
-import { remove } from '../services/comment.service';
+
+export const getCommentList = (req, res) => {
+  getList()
+    .then(data => {
+      if (!data) {
+        return res.status(404).json(data);
+      }
+      let commentList = [];
+      data.forEach(doc => {
```
Comment (jeffersoneagley) [inline server/handlers/comment.js]: `const commentList = _.map(data, doc=>...`

File: server/handlers/notification.js [hunk 2]
```diff
@@ -25,10 +21,14 @@ export const getNotification = async (req, res) => {
       let notification = [];
       data.forEach(doc => {
```
Comment (jeffersoneagley) [inline server/handlers/notification.js]: same as above

File: server/services/comment.service.js [hunk 3]
```diff
@@ -1,8 +1,22 @@
 import Comment from '../models/comment.model';
+import mongo from 'mongodb';
 
 import mongoConnection from '../util/mongo';
 mongoConnection();
 
+export const getList = async () => {
+  let commentList = [];
+  let error = {};
+  commentList = await Comment.find({})
```
Comment (jeffersoneagley) [inline server/services/comment.service.js]: const commentList = await blablabla

File: src/App.js [hunk 4]
```diff
@@ -51,19 +57,34 @@ if (token && handle) {
...
-            <AuthRoute path='/signup' component={signup} />
-            <Route path='/u/:handle' component={user} />
-          </Switch>
-        </div>
-      </Router>
-    </Provider>
+    <NotificationProvider getNotificationList={getNotificationList}>
+      <PostProvider fetchPostList={fetchPostList}>
+        <CommentProvider fetchCommentList={fetchCommentList}>
+          <Provider store={store}>
```
Comment (jeffersoneagley) [inline src/App.js]: Yo we heard you like providers

File: src/components/Navbar.js [hunk 5]
```diff
@@ -65,9 +59,11 @@ const ButtonRegister = ({ isLoggedIn }) => {
 const ButtonNotification = ({ isLoggedIn }) => {
   if (isLoggedIn) {
     return (
-      <CustomButton tip='Notifications'>
-        <Notifications />
-      </CustomButton>
+      <Link to='/notification'>
+        <CustomButton tip='Notifications'>
```
Comment (jeffersoneagley) [inline src/components/Navbar.js]: This is why it tabs twice over that bar

File: src/components/notification/index.js [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+          💬
+        </span>
+      </h1>
+      <hr />
+      <Content />
+    </div>
+  );
+};
+
+export default NotificationPanel;
```
Comment (jeffersoneagley) [inline src/components/notification/index.js]: You should use a named export by directly doing
`export const NotificationPanel = ({}) => ...`

Default exports break things.

File: src/components/notification/notificationCard/NotificationCardContent.js [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+    }
+    return (
+      <Typography color='primary'>
+        {notification?.sender} commented on your post!
+      </Typography>
+    );
+  };
+  return (
+    <CardContent className={classes?.content}>
+      {markNotificationNew()}
```
Comment (jeffersoneagley) [inline src/components/notification/notificationCard/NotificationCardContent.js]: `<NotificationNew/> `

File: src/components/notification/notificationCard/NotificationCardContent.js [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+      <span>{makeFromSenderType()}</span>
+      <span>"{makeNotificationContent()}"</span>
```
Comment (jeffersoneagley) [inline src/components/notification/notificationCard/NotificationCardContent.js]: `<SenderType/>`, and you can probably put the span in the return of SenderType

File: src/components/post/postList/postDialog/index.js [hunk 9]
```diff
@@ -37,13 +37,29 @@ const PostDialog = ({
...
+    if ((currentOldPath = currentNewPath)) {
+      currentOldPath = `/u/${userHandle}`;
+    }
+    window.history.pushState(null, null, currentNewPath);
     setOpen(true);
+    setOldPath(currentOldPath);
+    getPost(postId);
   };
   const handleClose = () => {
+    window.history.pushState(null, null, oldPath);
```
Comment (jeffersoneagley) [inline src/components/post/postList/postDialog/index.js]: `const history = useHistory()`

File: server/handlers/comment.js (hunk 1 above)
Comment (Ilithor) [reply server/handlers/comment.js]: Fixed

File: server/handlers/notification.js (hunk 2 above)
Comment (Ilithor) [reply server/handlers/notification.js]: Fixed

File: server/services/comment.service.js (hunk 3 above)
Comment (Ilithor) [reply server/services/comment.service.js]: Fixed

File: src/App.js (hunk 4 above)
Comment (Ilithor) [reply src/App.js]: Fixed

File: src/components/Navbar.js (hunk 5 above)
Comment (Ilithor) [reply src/components/Navbar.js]: Fixed

File: src/components/notification/index.js (hunk 6 above)
Comment (Ilithor) [reply src/components/notification/index.js]: Fixed

File: src/components/notification/notificationCard/NotificationCardContent.js (hunk 7 above)
Comment (Ilithor) [reply src/components/notification/notificationCard/NotificationCardContent.js]: Fixed

File: src/components/notification/notificationCard/NotificationCardContent.js (hunk 8 above)
Comment (Ilithor) [reply src/components/notification/notificationCard/NotificationCardContent.js]: Fixed

File: src/components/post/postList/postDialog/index.js (hunk 9 above)
Comment (Ilithor) [reply src/components/post/postList/postDialog/index.js]: Fixed

File: src/components/login/LoginForm.js [hunk 10]
```diff
@@ -12,7 +12,7 @@ import style from '../../style';
 /** View component for displaying the login form to the user
  * @type {ILoginFormComponentProps}
  */
-export const LoginForm = ({
+export const LoginFormView = ({
```
Comment (jeffersoneagley) [inline src/components/login/LoginForm.js]: Unneeded export of view component, unless you plan on testing the view directly

File: src/components/notification/notificationCard/index.js [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+ * @param {Post} props.post
+ * @param {UserComment} props.comment
+ * @param {Reac} history
+ */
+export const NotificationCardView = ({
+  classes = {},
+  notification = {},
+  post = {},
+  comment = {},
+  history,
```
Comment (jeffersoneagley) [inline src/components/notification/notificationCard/index.js]: `const history = useHistory()`

no reason to pass this guy

File: src/pages/user.js [hunk 12]
```diff
@@ -13,15 +13,17 @@ import { getUserPostAction } from '../redux/actions/userActions';
...
 
-const User = ({ match, data, getUserPostAction, getPostList }) => {
+const UserPageView = ({ match, data, getUserPostAction, getPostList }) => {
   const [profile, setProfile] = useState(null);
   const [postList, setPostList] = useState(null);
+  const [postIdParam, setPostIdParam] = useState(null);
   const handle = match.params.handle;
   useEffect(() => {
+    const postId = match.params.postId;
+    if (!!postId) {
```
Comment (jeffersoneagley) [inline src/pages/user.js]: I think there's a useParams or useMatch hook

File: src/components/login/LoginForm.js (hunk 10 above)
Comment (Ilithor) [reply src/components/login/LoginForm.js]: I currently has some basic tests dependent on this export

File: src/components/notification/notificationCard/index.js (hunk 11 above)
Comment (Ilithor) [reply src/components/notification/notificationCard/index.js]: Fixed

File: src/pages/user.js (hunk 12 above)
Comment (Ilithor) [reply src/pages/user.js]: Fixed

**Pull request merged**